File: ItTraining/WebContent/it-training-angular/it-training/src/app/entities/apprenant.ts
import { Adresse } from "./adresse";
import { Personne } from "./personne";
import { Module } from "./module";
import { Machine } from "./machine";
import { Formation } from "./formation";
//import { Entreprise } from "./entreprise";

export class Apprenant extends Personne {

    public id: number;
    public candidatAccepte: boolean;
    public formation: Formation;
    public machine: Machine;
    public entreprise: Entreprise;
=======
    
    public candidatAccepte: boolean;
    public formation: Formation;
    public machine: Machine;

>>>>>>> 16e153f3859b5daeff356c767d267be414d57f8b
}

File: ItTraining/src/fr/ib/bo/Apprenant.java
package fr.ib.bo;


import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;

import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@DiscriminatorValue(value="A")
public class Apprenant extends Personne {
	private boolean candidatAccepte;
	
	@ManyToMany(cascade = CascadeType.PERSIST)
	@JoinColumn(name="formation")
	@JsonIgnore
	private List<Formation> formation;
	
	@OneToOne(cascade = CascadeType.PERSIST)
	@JoinColumn(name="machine")
	private Machine machine;
	
	@ManyToOne(cascade = CascadeType.PERSIST)
	@JoinColumn(name="entreprise")
	private Entreprise entreprise;
	
	
	//Construsteurs
	public Apprenant() {
		super();
	}



	public boolean isCandidatAccepte() {
		return candidatAccepte;
	}

	public void setCandidatAccepte(boolean candidatAccepte) {
		this.candidatAccepte = candidatAccepte;
	}

	

	public List<Formation> getFormation() {
		return formation;
	}



	public void setFormation(List<Formation> formation) {
		this.formation = formation;
	}



	public Machine getMachine() {
		return machine;
	}

	public void setMachine(Machine machine) {
		this.machine = machine;
	}

	public Entreprise getEntreprise() {
		return entreprise;
	}

	public void setEntreprise(Entreprise entreprise) {
		this.entreprise = entreprise;
	}
	
	
	//Getters & Setters 

}
